test/functional/ui/messages2_spec.lua

```diff
@@ -253,4 +253,125 @@ describe('messages2', function()
       {16::}^                                                    |
     ]])
   end)
+
+  it('paging prompt dialog #35191', function()
+    local top = [[
+                                                           |
+      {1:~                                                    }|*3
+      {3:─────────────────────────────────────────────────────}|
+      0                                                    |
+      1                                                    |
+      2                                                    |
+      3                                                    |
+      4                                                    |
+      5                                                    |
+      6 [+93]                                              |
+      Type number and <Enter> or click with the mouse (q or|
+       empty cancels): ^                                    |
+    ]]
+    feed(':call inputlist(range(100))<CR>')
+    screen:expect(top)
+    feed('j')
+    screen:expect([[
+                                                           |
+      {1:~                                                    }|*3
+      {3:─────────────────────────────────────────────────────}|
+      1 [+1]                                               |
+      2                                                    |
+      3                                                    |
+      4                                                    |
+      5                                                    |
+      6                                                    |
+      7 [+92]                                              |
```

Comment:
Getting this to work was a bit tricky/messy. And representing a scroll bar like this may be a bit arbitrary, but we are already using it to indicate spilled lines in the cmdline. It seems more useful to me than the current message grid behavior, which only allows scrolling until the cmdline prompt is reached, with the text `--More--` at the bottom.